### PR TITLE
qt: Delete wallet views when wallets are removed in removeAllWallets

### PR DESCRIPTION
**Pull request by UdjinM6**
This fixes a qt wallet crash on shutdown.

### CODE REVIEW

File: src/qt/walletframe.cpp [hunk 1]
```diff
@@ -99,8 +99,10 @@ bool WalletFrame::removeWallet(const QString &name)
 void WalletFrame::removeAllWallets()
 {
     QMap<QString, WalletView*>::const_iterator i;
-    for (i = mapWalletViews.constBegin(); i != mapWalletViews.constEnd(); ++i)
+    for (i = mapWalletViews.constBegin(); i != mapWalletViews.constEnd(); ++i) {
         walletStack->removeWidget(i.value());
+        delete i.value();
```
Comment (xdustinface) [inline src/qt/walletframe.cpp:104]:
```suggestion
       i.value()->deleteLater();
```

Not sure if it matters at all when its removed from the parent, but maybe like ^ ?

**Review by xdustinface — APPROVED** [review]
utACK with one maybe below

File: src/qt/walletframe.cpp (hunk 1 above)
Comment (UdjinM6) [reply src/qt/walletframe.cpp:104]: `removeAllWallets` is called from `requestShutdown` https://github.com/dashpay/dash/blob/develop/src/qt/dash.cpp#L431 which is called outside of the main event loop https://github.com/dashpay/dash/blob/develop/src/qt/dash.cpp#L818. It doesn't look like a good place to use `deleteLater` unless I'm misreading this https://doc.qt.io/qt-5/qobject.html#deleteLater.

**Review by PastaPastaPasta — APPROVED** [review]
utACK

**Pull request merged**